AWSIoTPythonSDK/core/protocol/mqtt_core.py

```diff
@@ -190,13 +190,16 @@ def configure_draining_interval_sec(self, draining_interval_sec):
         self._logger.info("Configuring offline requests queue draining interval: %f sec", draining_interval_sec)
         self._event_consumer.update_draining_interval_sec(draining_interval_sec)
 
-    def connect(self, keep_alive_sec):
+    def connect(self, keep_alive_sec, raise_on_timeout=True):
         self._logger.info("Performing sync connect...")
         event = Event()
         self.connect_async(keep_alive_sec, self._create_blocking_ack_callback(event))
         if not event.wait(self._connect_disconnect_timeout_sec):
-            self._logger.error("Connect timed out")
-            raise connectTimeoutException()
+            self._logger.warning("Connect timed out, will try to reconnect")
```

Comment:
This is not really an error as the background_network_io will still try to reconnect later